hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,195 @@
+    bool appendFreeSpaceJointPoint();
+    void planProcessPath();
+
+    // void HybridPlanner::appendDescartesCartPoint(EigenSTL::vector_Affine3d& pose);
+
+    /* Create an axial trajectory pt from a given tf transform */
+    descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization);   
+
+    /*Translates a point relative to a reference pose to an absolute transformation
+    Also flips the z axis of the orientation to point INTO the plane specified by the
+    reference frame.*/
+    tf::Transform createNominalTransform(const geometry_msgs::Pose& ref_pose, const geometry_msgs::Point& point);
+
+    void fromDescartesToMoveitTrajectory(const descartes_traj& in_traj);
+
+
+    void publishPosesMarkers(const EigenSTL::vector_Affine3d& poses);
+
+
+  protected:
+
+    /* Data
+     * Holds the data used by the various functions  
+     */
+    HybridConfiguration config_;
+
+    /* ROS Constructs
+     * Components needed to successfully run a ros-node and perform other important
+     * ros-related tasks
+     */
+    ros::NodeHandle nh_;                        /* Object used for creating and managing ros resources*/
+    ros::Publisher marker_publisher_;           /* Publishes visualization message to Rviz */
+    ros::ServiceClient moveit_run_path_client_; /* Sends a robot trajectory to moveit for execution */
+
+    /* Descartes Constructs
+     * Components accessing the semi-constrained path planning capabilities in the Descartes library
+     */
+    descartes_traj& descartes_traj_input_;
+    descartes_traj& descartes_traj_output_;  /* Maintains the semi-constrained points and segments*/
+    trajectory_msgs::JointTrajectory& joint_traj_descartes_;//TODO change name
+    descartes_core::RobotModelPtr descartes_robot_model_ptr_; /* Performs tasks specific to the Robot
+                                                       such IK, FK and collision detection*/
+    descartes_planner::SparsePlanner sparse_planner_;      /* Plans a smooth robot path given a trajectory of points */
+    descartes_planner::DensePlanner dense_planner_;
+    int no_of_descartes_segments_;
+
+    /* MoveIt! Constructs
+     * Components accesing the free-space planning capabilities in the MoveIt! library.
+     */
+    moveit::core::RobotStatePtr kinematic_state_;
+    robot_model::RobotModelConstPtr& kinematic_model_;
+    const moveit::core::JointModelGroup* joint_model_group_; //TOCHECK what all to keep const coz of return type of kinematic_model_->getJointModelGroup
+    boost::shared_ptr<move_group_interface::MoveGroup> moveit_group_ptr_;
+    moveit_robot_traj moveit_robot_traj_;
+    robot_trajectory::RobotTrajectory overall_robot_traj_; //TODO better non-confusing names
+    move_group_interface::MoveGroup::Plan overall_plan_; //TOCHECK Member variable vs instantiate within the plan or addFreeSpaceSegment function?
+    int no_of_free_segments_; 
+    //TODO just one naming scheme to avoid confusion 
```

Comment:
Could you point me to the reference in Godel that you are referring to?
How about we start with the simple case of free-motion -> process path -> free motion?